src/Kiota.Builder/Writers/CSharp/CodePropertyWriter.cs

```diff
@@ -36,6 +36,8 @@ private void WritePropertyInternal(CodeProperty codeElement, LanguageWriter writ
         if (codeElement.Parent is not CodeClass parentClass) throw new InvalidOperationException("The parent of a property should be a class");
         var backingStoreProperty = parentClass.GetBackingStoreProperty();
         var setterAccessModifier = codeElement.ReadOnly && codeElement.Access > AccessModifier.Private ? "private " : string.Empty;
+        var nullableEx = codeElement.IsOfKind(CodePropertyKind.AdditionalData) ? " ?? throw new InvalidOperationException(\"AdditionalData can not be null\")" : string.Empty;

```

Comment:
Does it make sense to move these two lines to line 54 in the `case CodePropertyKind.Custom when backingStoreProperty != null:` case block? I think these variables are really only used applicable in that case block and need not be computed for every codeElement kind.

Reply:
@Balkoth  I believe the `MapsAdditionalDataPropertiesToBackingStore` test will need refactoring as well givent the changed output.

Reply:
I already had it there, moved it because the other stuff is also declared at the top. But it makes more sense to move it down.

Reply:
Fixed.